satpy/resample.py

```diff
@@ -509,27 +472,117 @@ def compute(self, data, fill_value=None, **kwargs):
         return res
 
 
+class NativeResampler(BaseResampler):
+    """Expand or reduce input datasets to be the same shape.
+
+    If `expand=True` (default) input datasets are replicated in both
+    dimensions to be the same as the shape provided on initialization.
+
+    If `expand=False` then input datasets are strided to the shape
+    provided.
+
+    """
+    def resample(self, data, cache_dir=False, mask_area=False, **kwargs):
+        # use 'mask_area' with a default of False. It wouldn't do anything.
+        return super(NativeResampler, self).resample(data,
+                                                     cache_dir=cache_dir,
+                                                     mask_area=mask_area,
+                                                     **kwargs)
+
+    @staticmethod
+    def expand_reduce(d_arr, repeats, axis):
+        if not isinstance(d_arr, da.Array):
+            d_arr = da.from_array(d_arr, chunks=CHUNK_SIZE)
+        if repeats == 1:
+            return d_arr
+        elif repeats > 1:
+            if not repeats.is_integer():
+                raise ValueError("Expand factor must be a whole number")
+            # expand
+            return da.repeat(d_arr, int(repeats), axis=axis)
+        else:
+            # reduce
+            repeats = 1. / repeats
+            if not repeats.is_integer():
+                raise ValueError("Reduce factor must be a whole number")
+            slices = tuple(slice(None, None, int(repeats))
+                           if a == axis else slice(None)
+                           for a in range(d_arr.ndim))
+            return d_arr[slices]
```

Comment:
I would like to have aggregation here...

Reply:
Yeah, you're right. It just screws up the simplicity of the whole thing, but ok.

Reply:
FYI I don't think this can be done with the current way that I know how to do it unless the inputs are limited to 2D arrays. I'm using `map_blocks` and if you add any more dimensions there is a chance that the block I'm getting isn't the right shape. Or rather, I don't know what axes I'm getting. I'm just going to require 2D for now and if someone needs something else I'll worry about it then.